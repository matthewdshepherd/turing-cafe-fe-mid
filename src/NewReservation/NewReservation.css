.NewReservation {
  display: flex;
  justify-content: space-evenly;
  color: white;
  width: 100%;
  align-items: flex-end;
}
.labelInput__div {
  display: flex;
  flex-direction: column;
  align-items: flex-start;

}

.makeReservationButton {
  height: 50%;
  background-color: teal;
  color: white;

}

.NewReservationInput,
.makeReservationButton {
}