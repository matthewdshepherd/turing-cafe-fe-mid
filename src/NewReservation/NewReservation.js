import React from 'react'
import './NewReservation.css'

export class NewReservation extends React.Component {
  constructor() {
    super()
      this.state = {
        name: '',
        date: '',
        time: '',
        number: null
      }
    
  }

  handlChange = event => {
    console.log(event)
    this.setState({ [event.target.name]: event.target.value })
  }

  handleSubmit = event => {
    event.preventDefault();
    const { addReservation } = this.props;
    const newReservation = { ...this.state, id: date.Now()};
    addReservation(newReservation)
    this.resetInputs();
  }

  resetInputs = () => {
    this.setState({
      name: '',
      date: '',
      time: '',
      number: null
    })
  }

  render() {
    return (
      <form className={'NewReservation'}>
        <div className={'labelInput__div'}>
          <label htmlFor='name' className='NewReservationLabel'>Name</label>
          <input
          className='NewReservationInput'
          id='name'
          type='text'
          name='name'
          placeholder='Name'
          onChange={this.handlChange}
          value={this.state.name} />
        </div>
        <div className={'labelInput__div'}>
          <label htmlFor='date' className='NewReservationLabel'>Date</label>
          <input
          className='NewReservationInput'
          id='date'
          type='text'
          name='date'
          placeholder='Date'
          onChange={this.handlChange}
          value={this.state.date} />
        </div>
        <div className={'labelInput__div'}>
          <label htmlFor='time' className='NewReservationLabel'>Time</label>
          <input
          className='NewReservationInput'
          id='time'
          type='time'
          name='time'
          placeholder='Time'
          onChange={this.handlChange}
          value={this.state.time} />
        </div>
        <div className={'labelInput__div'}>
         <label htmlFor='numOfGuests' className='NewReservationLabel'>Number Of Guests</label>
          <input
          className='NewReservationInput'
          id='numOfGuests'
          type='number'
          name='numOfGuests'
          placeholder='Number Of Guests'
          onChange={this.handlChange}
          value={this.state.number} />
        </div>
        <button className={'makeReservationButton'}>Make Reservation</button>
      </form>
    )
  }
}

export default NewReservation;