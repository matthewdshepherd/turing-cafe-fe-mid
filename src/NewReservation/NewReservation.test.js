import React from 'react';
import { shallow } from 'enzyme';
import NewReservation from './NewReservation'

describe('NewReservation', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewReservation
      addReservation={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  });
})