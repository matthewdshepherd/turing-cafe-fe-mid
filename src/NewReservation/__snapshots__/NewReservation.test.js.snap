// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewReservation should match the snapshot with all data passed in correctly 1`] = `
<form
  className="NewReservation"
>
  <div
    className="labelInput__div"
  >
    <label
      className="NewReservationLabel"
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="NewReservationInput"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
  </div>
  <div
    className="labelInput__div"
  >
    <label
      className="NewReservationLabel"
      htmlFor="date"
    >
      Date
    </label>
    <input
      className="NewReservationInput"
      id="date"
      name="date"
      onChange={[Function]}
      placeholder="Date (mm/dd)"
      type="text"
      value=""
    />
  </div>
  <div
    className="labelInput__div"
  >
    <label
      className="NewReservationLabel"
      htmlFor="time"
    >
      Time
    </label>
    <input
      className="NewReservationInput"
      id="time"
      name="time"
      onChange={[Function]}
      placeholder="Time"
      type="time"
      value=""
    />
  </div>
  <div
    className="labelInput__div"
  >
    <label
      className="NewReservationLabel"
      htmlFor="number"
    >
      Number Of Guests
    </label>
    <input
      className="NewReservationInput"
      id="number"
      name="number"
      onChange={[Function]}
      placeholder="Number Of Guests"
      type="text"
      value=""
    />
  </div>
  <button
    className="makeReservationButton"
    onClick={[Function]}
    type="button"
  >
    Make Reservation
  </button>
</form>
`;
